src/Security/Authentication/FrontendAuthenticationSuccessHandler.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * Copyright (c) 2005-2017 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+namespace Contao\CoreBundle\Security\Authentication;
+
+use Contao\FrontendUser;
+use Contao\PageModel;
+use Contao\StringUtil;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Http\ParameterBagUtils;
+
+class FrontendAuthenticationSuccessHandler extends AuthenticationSuccessHandler
+{
+    /**
+     * @var TokenInterface
+     */
+    private $user;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function onAuthenticationSuccess(Request $request, TokenInterface $token): RedirectResponse
+    {
+        $this->user = $token->getUser();
+
+        return parent::onAuthenticationSuccess($request, $token);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function determineTargetUrl(Request $request)
+    {
+        if (!$this->user instanceof FrontendUser) {
+            return parent::determineTargetUrl($request);
+        }
+
+        if (ParameterBagUtils::getRequestParameterValue($request, '_always_use_target_path')
```

Comment:
You are using an internal method here.

Reply:
I know. I'm using the same method Symfony does in their implementation…

Reply:
Except its their method so they are "allowed" to use it. Is there no other way to retrieve the value?